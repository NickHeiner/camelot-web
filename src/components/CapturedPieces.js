import {Glyphicon, Badge} from 'react-bootstrap';
import React, {PureComponent} from 'react';

export default class CapturedPieces extends PureComponent {
    render() {
        // Is this really necessary or is it an artifact of some other bad practice?
        if (!this.props.gameState) {
            return null;
        }

        const whichPlayer = this.props.whosePiecesWereCaptured === 'host' ? 'playerA' : 'playerB';
        return <div className="captured">
            <div className="piece-count-pair">
                <Glyphicon glyph="tower" className={this.props.whosePiecesWereCaptured} /> 
                <Badge>{this.props.gameState.capturedPieces[whichPlayer].knight}</Badge> 
            </div>
            <div className="piece-count-pair">
                <Glyphicon glyph="pawn" className={this.props.whosePiecesWereCaptured} /> 
                <Badge>{this.props.gameState.capturedPieces[whichPlayer].pawn}</Badge>
            </div>
        </div>;
    }
}
