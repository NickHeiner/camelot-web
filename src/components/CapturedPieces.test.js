import CapturedPieces from './CapturedPieces';
import React from 'react';
import reactTestRenderer from 'react-test-renderer';
import camelotEngine from 'camelot-engine';
import {fromJS} from 'immutable';

/* eslint-disable no-magic-numbers */

describe('CapturedPieces', () => {
  it('renders when there is no game state', () => {
    expect(reactTestRenderer.create(
            <CapturedPieces />
        )).toMatchSnapshot();
  });

  const gameState = fromJS(camelotEngine().createEmptyGame());

  describe('with game state', () => {
    describe('no captured pieces', () => {
      it('host is captured', () => {
        expect(reactTestRenderer.create(
                    <CapturedPieces gameState={gameState.toJS()} whosePiecesWereCaptured="host" />
                )).toMatchSnapshot();
      });
      it('opponent is captured', () => {
        expect(reactTestRenderer.create(
                    <CapturedPieces gameState={gameState.toJS()} whosePiecesWereCaptured="opponent" />
                )).toMatchSnapshot();
      });
    });

    describe('captured pieces', () => {
      it('host is captured', () => {
        const withCaptures = gameState
          .setIn(['capturedPieces', 'playerA', 'knight'], 1)
          .setIn(['capturedPieces', 'playerA', 'pawn'], 3);
        expect(reactTestRenderer.create(
          <CapturedPieces gameState={withCaptures.toJS()} whosePiecesWereCaptured="host" />
        )).toMatchSnapshot();
      });
      it('opponent is captured', () => {
        const withCaptures = gameState
          .setIn(['capturedPieces', 'playerB', 'knight'], 4)
          .setIn(['capturedPieces', 'playerB', 'pawn'], 2);
        expect(reactTestRenderer.create(
          <CapturedPieces gameState={withCaptures.toJS()} whosePiecesWereCaptured="opponent" />
        )).toMatchSnapshot();
      });
    });
  });
});
