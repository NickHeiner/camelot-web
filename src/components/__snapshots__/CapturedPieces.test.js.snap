exports[`CapturedPieces renders when there is no game state 1`] = `null`;

exports[`CapturedPieces with game state captured pieces host is captured 1`] = `
<div
  className="captured">
  <div
    className="piece-count-pair">
    <span
      className="host glyphicon glyphicon-tower" />
    <span
      className="badge">
      0
    </span>
  </div>
  <div
    className="piece-count-pair">
    <span
      className="host glyphicon glyphicon-pawn" />
    <span
      className="badge">
      0
    </span>
  </div>
</div>
`;

exports[`CapturedPieces with game state captured pieces opponent is captured 1`] = `
<div
  className="captured">
  <div
    className="piece-count-pair">
    <span
      className="opponent glyphicon glyphicon-tower" />
    <span
      className="badge">
      0
    </span>
  </div>
  <div
    className="piece-count-pair">
    <span
      className="opponent glyphicon glyphicon-pawn" />
    <span
      className="badge">
      0
    </span>
  </div>
</div>
`;

exports[`CapturedPieces with game state no captured pieces host is captured 1`] = `
<div
  className="captured">
  <div
    className="piece-count-pair">
    <span
      className="host glyphicon glyphicon-tower" />
    <span
      className="badge">
      0
    </span>
  </div>
  <div
    className="piece-count-pair">
    <span
      className="host glyphicon glyphicon-pawn" />
    <span
      className="badge">
      0
    </span>
  </div>
</div>
`;

exports[`CapturedPieces with game state no captured pieces opponent is captured 1`] = `
<div
  className="captured">
  <div
    className="piece-count-pair">
    <span
      className="opponent glyphicon glyphicon-tower" />
    <span
      className="badge">
      0
    </span>
  </div>
  <div
    className="piece-count-pair">
    <span
      className="opponent glyphicon glyphicon-pawn" />
    <span
      className="badge">
      0
    </span>
  </div>
</div>
`;
